src/pymor/reductors/aaa.py

```diff
@@ -0,0 +1,391 @@
+# This file is part of the pyMOR project (https://www.pymor.org).
+# Copyright pyMOR developers and contributors. All rights reserved.
+# License: BSD 2-Clause License (https://opensource.org/licenses/BSD-2-Clause)
+
+import itertools
+
+import numpy as np
+
+from pymor.core.base import BasicObject
+from pymor.models.transfer_function import TransferFunction
+
+
+class pAAAReductor(BasicObject):
+    """Reductor implementing the parametric AAA algorithm.
+
+    The reductor implements the parametric AAA algorithm and can be used either with
+    data or a given full-order model which has a `transfer_function` attribute. MIMO
+    data is accepted.
+
+    Parameters
+    ----------
+    sampling_values
+        Values where sample data has been evaluated or the full-order model should be evaluated.
+        Sampling values are represented as a nested list `svs` such that `svs[i]`corresponds
+        to sampling values of the `i-th` variable.
+    samples_or_fom
+        Can be either a full-order model (|TransferFunction| or |Model| with a `transfer_function`
+        attribute) or data sampled at the values specified in `sampling_values`. Samples are
+        represented as a tensor `S`. E.g., for 3 inputs `S[i,j,k]` corresponds to the sampled
+        value at `(sampling_values[0][i],sampling_values[1][j],sampling_values[2][k])`. In the
+        MIMO case `S[i,j,k]` represents a matrix of dimension `dim_output` times `dim_input`.
+    conjugate
+        Whether to compute complex conjugates of first sampling variables and enforce
+        interpolation in complex conjugate pairs (allows for constructing real system matrices).
+    nsp_tol
+        Tolerance for null space of higher-dimensional Loewner matrix to check for
+        interpolation or convergence.
+    post_process
+        Whether to do post-processing or not. If the Loewner matrix has a null space
+        of dimension greater than 1, it is assumed that the algorithm converged to
+        a non-minimal order interpolant which may cause numerical issues. In this case,
+        the post-processing procedure computes an interpolant of minimal order.
+    L_rk_tol
+        Tolerance for ranks of 1-D Loewner matrices computed in post-processing.
+    """
+
+    def __init__(self, sampling_values, samples_or_fom, conjugate=True, nsp_tol=1e-16, post_process=True,
+                 L_rk_tol=1e-8):
+        if isinstance(samples_or_fom, TransferFunction) or hasattr(samples_or_fom, 'transfer_function'):
+            fom = samples_or_fom
+            if not isinstance(samples_or_fom, TransferFunction):
+                fom = fom.transfer_function
+            self.num_vars = 1 + len(fom.parameters)
```

Comment:
Some parameters can be higher-dimensional, so `fom.parameters.dim` might be better than `len`.